packages/signature-controller/src/types.ts

```diff
@@ -71,11 +77,29 @@ export type MessageParamsTyped = MessageParams & {
         primaryType: string;
         message: Json;
       };
-
+  /** Signature method V1, V3 or V4 */
+  signatureMethod?: string;
   /** Version of the signTypedData request. */
   version?: string;
 };
 
+export type DecodedRequestInfo =
+  | {
+      assetType: string;
+      changeType: string;
+      address: string;
+      amount: string;
+      contractAddress: string;
+    }
+  | {
+      error: {
+        message: string;
+        type: string;
+      };
+    }
+  | 'IN_PROGRESS'
```

Comment:
As mentioned elsewhere, is it cleaner to track this via an explicit `decodingLoading` property on the `SignatureRequest`?

Reply:
Fixed